dali/pipeline/data/tensor_vector_test.cc

```diff
@@ -52,6 +59,633 @@ constexpr cudaStream_t cuda_stream = 0;
+  for (auto b : buffer) {
+    EXPECT_EQ(b, number);
+  }
+}
+
+template <typename Backend>
+Tensor<Backend> ReturnTvAsTensor(uint8_t number) {
+  TensorVector<Backend> tv;
+  tv.SetContiguous(true);
+  tv.Resize(uniform_list_shape(4, {1, 2, 3}), DALI_UINT8);
+  tv.SetLayout("HWC");
+  FillWithNumber(tv, number);
+  return tv.AsTensor();
+}
+
+}  // namespace
+
+TYPED_TEST(TensorVectorSuite, ResizeSetSize) {
+  constexpr bool is_device = std::is_same_v<TypeParam, GPUBackend>;
+  const auto order = is_device ? AccessOrder(cuda_stream) : AccessOrder::host();
+  TensorVector<TypeParam> tv;
+  tv.set_pinned(false);
+  tv.set_order(order);
+  tv.set_sample_dim(2);
+  tv.SetLayout("XY");
+  tv.SetContiguous(BatchState::Contiguous);
+  tv.SetSize(3);
+
+
+  auto empty_2d = TensorShape<>{0, 0};
+  for (int i = 0; i < 3; i++) {
+    EXPECT_EQ(tv[i].raw_data(), nullptr);
+    EXPECT_EQ(tv[i].shape(), empty_2d);
+    EXPECT_EQ(tv[i].type(), DALI_NO_TYPE);
+  }
+
+  // Setting just the type
+  tv.set_type(DALI_INT32);
+  for (int i = 0; i < 3; i++) {
+    EXPECT_EQ(tv[i].raw_data(), nullptr);
+    EXPECT_EQ(tv[i].shape(), empty_2d);
+    EXPECT_EQ(tv[i].type(), DALI_INT32);
+  }
+  auto new_shape = TensorListShape<>{{1, 2, 3}, {2, 3, 4}, {3, 4, 5}};
+  tv.Resize(new_shape);
+  tv.SetLayout("HWC");
+
+  const auto *base = static_cast<const uint8_t*>(unsafe_sample_owner(tv, 0).get());
+
+  for (int i = 0; i < 3; i++) {
+    EXPECT_EQ(tv[i].raw_data(), base);
+    EXPECT_EQ(tv[i].shape(), new_shape[i]);
+    EXPECT_EQ(tv[i].type(), DALI_INT32);
+    base += sizeof(int32_t) * new_shape[i].num_elements();
+  }
+
+  tv.SetSize(4);
+  EXPECT_TRUE(tv.IsContiguous());
+
+  auto empty_3d = TensorShape<>{0, 0, 0};
+
+  // New one should be empty
+  EXPECT_EQ(tv[3].raw_data(), nullptr);
+  EXPECT_EQ(tv[3].shape(), empty_3d);
+  EXPECT_EQ(tv[3].type(), DALI_INT32);
+
+  tv.SetSize(2);
+  tv.SetSize(3);
+  EXPECT_TRUE(tv.IsContiguous());
+
+  base = static_cast<const uint8_t*>(unsafe_sample_owner(tv, 0).get());
+
+  for (int i = 0; i < 2; i++) {
+    EXPECT_EQ(tv[i].raw_data(), base);
+    EXPECT_EQ(tv[i].shape(), new_shape[i]);
+    EXPECT_EQ(tv[i].type(), DALI_INT32);
+    base += sizeof(int32_t) * new_shape[i].num_elements();
+  }
+
+  // As we are contiguous, thus we are sharing the contiguous buffer via every sample,
+  // the share is removed and no allocation should happen when we made the size bigger
+  for (int i = 2; i < 3; i++) {
+    EXPECT_EQ(tv[i].raw_data(), nullptr);
+    EXPECT_EQ(tv[i].shape(), empty_3d);
+    EXPECT_EQ(tv[i].type(), DALI_INT32);
+  }
+}
+
+
+TYPED_TEST(TensorVectorSuite, NoForcedChangeContiguousToNoncontiguous) {
+  constexpr bool is_device = std::is_same_v<TypeParam, GPUBackend>;
+  TensorVector<TypeParam> tv;
+  tv.SetContiguous(BatchState::Contiguous);
+  auto new_shape = TensorListShape<>{{1, 2, 3}, {2, 3, 4}, {3, 4, 5}};
+  EXPECT_THROW(tv.Resize(new_shape, DALI_FLOAT, BatchState::Noncontiguous), std::runtime_error);
```

Comment:
why does this fail?

Reply:
We forced/pinned the tv to be contiguous, and we are trying to resize it as noncontiguous.

Reply:
A short message would help the reader in the future. So SetContiguous(...) "forces" the state so that it can't be changed later by Resize. Right?

Reply:
Yes.

Reply:
done